alibi/explainers/tests/test_partial_dependence.py

```diff
@@ -467,21 +487,471 @@ def test_sklearn_blackbox(rf_classifier, adult_data, features, params):
                                      **params)
 
     # compute alibi explanation
-    explainer = PartialDependence(predictor=predictor_pipeline.predict_proba,
-                                  feature_names=adult_data['metadata']['feature_names'],
-                                  categorical_names=adult_data['metadata']['category_map'])
-    exp_alibi = explainer.explain(X=X_train,
-                                  features=features,
-                                  kind=params['kind'],
-                                  percentiles=params['percentiles'],
-                                  grid_resolution=params['grid_resolution'])
+    exp_alibi = get_alibi_pd_explanation(predictor=predictor_pipeline.predict_proba,
+                                         feature_names=adult_data['metadata']['feature_names'],
+                                         categorical_names=adult_data['metadata']['category_map'],
+                                         X=X_train,
+                                         features=features,
+                                         kind=params['kind'],
+                                         percentiles=params['percentiles'],
+                                         grid_resolution=params['grid_resolution'])
 
     # compare explanations
-    assert np.allclose(exp_alibi.pd_values[0][1], exp_sklearn['average'])
-    assert np.allclose(exp_alibi.ice_values[0][1], exp_sklearn['individual'])
+    assert np.allclose(exp_alibi.data['ice_values'][0][1], exp_sklearn['individual'])
+    assert np.allclose(exp_alibi.data['pd_values'][0][1], exp_sklearn['average'])
+    compare_feature_values(exp_alibi=exp_alibi, exp_sklearn=exp_sklearn)
+
+
+@pytest.fixture(scope='function')
```

Comment:
This seems to be just static data, is there a reason this has to be function scoped?

Reply:
Ah I see it is modified by test functions. Perhaps leave a comment here saying why this needs to be function scoped. Although probably would be better to use a parametrised fixture.

Reply:
Can we just use this pattern? https://docs.pytest.org/en/6.2.x/fixture.html#parametrizing-fixtures